### PR TITLE
Hw jdbc solution

### PR DESCRIPTION
**Pull request by BusyhinD**

### CODE REVIEW

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all manufacturers from DB", e);
+        }
+        return manufacturers;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        Manufacturer manufacturer = null;
+        String getManufacturerRequest
```
Comment (okuzan) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:63]: in this context it's more like `query` than `request`

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    private static final int FIRST_VARIABLE = 1;
+    private static final int SECOND_VARIABLE = 2;
+    private static final int THIRD_VARIABLE = 3;
+    private static final int ID_COLUMN = 1;
```
Comment (okuzan) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:21]: it's okay to use these integers as is for statement placement, it's not exactly reusing constants - every query has different parameters and order of parameters

File: src/main/java/mate/jdbc/Main.java [hunk 3]
```diff
@@ -1,7 +1,27 @@
...
+        manufacturer.setName("Opel");
+        manufacturer.setCountry("Germany");
+        Manufacturer manufacturerWithId = manufacturerDao.create(manufacturer);
+        manufacturerWithId.setName("Fiat");
+        manufacturerWithId.setCountry("Italy");
+        manufacturerDao.update(manufacturerWithId);
+        List<Manufacturer> manufacturers = manufacturerDao.getAll();
+        Optional<Manufacturer> manufacturerOptional =
+                manufacturerDao.get(manufacturerWithId.getId());
+        boolean isDeleted = manufacturerDao.delete(manufacturerWithId.getId());
```
Comment (okuzan) [inline src/main/java/mate/jdbc/Main.java:25]: this boolean is never checked

File: src/main/java/mate/jdbc/Main.java [hunk 4]
```diff
@@ -1,7 +1,27 @@
...
+                (ManufacturerDao) injector.getInstance(ManufacturerDao.class);
+        Manufacturer manufacturer = new Manufacturer();
+        manufacturer.setName("Opel");
+        manufacturer.setCountry("Germany");
+        Manufacturer manufacturerWithId = manufacturerDao.create(manufacturer);
+        manufacturerWithId.setName("Fiat");
+        manufacturerWithId.setCountry("Italy");
+        manufacturerDao.update(manufacturerWithId);
+        List<Manufacturer> manufacturers = manufacturerDao.getAll();
+        Optional<Manufacturer> manufacturerOptional =
```
Comment (okuzan) [inline src/main/java/mate/jdbc/Main.java:23]: does optional contain value? show that it's not empty

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            if (resultSet.next()) {
+                manufacturer = manufacturerBuilder(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get manufacturer from DB by ID " + id, e);
+        }
+        Optional<Manufacturer> resultOfQuery = Optional.ofNullable(manufacturer);
+        resultOfQuery.orElseThrow(()
+                -> new RuntimeException("Can't get manufacturer by id " + id));
+        return resultOfQuery;
```
Comment (Ivan95kos) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:75]: You don't have to throw an exception just return Optional.
This exception is thrown in main.
```suggestion
        return Optional.ofNullable(manufacturer);
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        Manufacturer manufacturer = null;
+        String getManufacturerQuery
+                = "SELECT * FROM manufacturers WHERE id = ? AND is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturerStatement = connection.prepareStatement(
+                        getManufacturerQuery, Statement.RETURN_GENERATED_KEYS)) {
```
Comment (Ivan95kos) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:63]: Use Statement.RETURN_GENERATED_KEYS only in create statement, it’s not needed in other methods.

File: src/main/java/mate/jdbc/util/ConnectionUtil.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Properties;
+
+public class ConnectionUtil {
+    static {
+        try {
+            Class.forName("com.mysql.cj.jdbc.Driver");
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException("Can't load JDBC driver for MySQL", e);
```
Comment (Ivan95kos) [inline src/main/java/mate/jdbc/util/ConnectionUtil.java:13]:
```suggestion
            throw new RuntimeException("Can't load JDBC driver for MySQL ", e);
```

File: src/main/java/mate/jdbc/util/ConnectionUtil.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+    public static Connection getConnection() {
+        try {
+            Properties dbProperties = new Properties();
+            dbProperties.put("user", "root");
+            dbProperties.put("password", "6913");
+            return DriverManager.getConnection(
+                    "jdbc:mysql://localhost:3306/taxi_service_db", dbProperties);
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't create connection to DB", e);
```
Comment (Ivan95kos) [inline src/main/java/mate/jdbc/util/ConnectionUtil.java:25]:
```suggestion
            throw new RuntimeException("Can't create connection to DB ", e);
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+        String getAllManufacturerQuery = "SELECT * FROM manufacturers WHERE is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllManufacturersStatement =
+                        connection.prepareStatement(getAllManufacturerQuery)) {
+            ResultSet resultSet = getAllManufacturersStatement.executeQuery();
+            while (resultSet.next()) {
+                manufacturers.add(manufacturerBuilder(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all manufacturers from DB", e);
```
Comment (Ivan95kos) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:51]:
```suggestion
            throw new DataProcessingException("Can't get all manufacturers from DB ", e);
```

**Review by Ivan95kos — CHANGES_REQUESTED** [review]

**Review by Ivan95kos — APPROVED** [review]
Great work

**Pull request closed without merging**